src/gui_common/TransitionManager.cs

```diff
@@ -36,6 +37,7 @@ private TransitionManager()
     public List<ITransition> TransitionSequence { get; } = new List<ITransition>();
 
     public bool HasQueuedTransitions => TransitionSequence.Count > 0;
+    public bool HasTransitionInProgress => TransitionSequence.Any(ts => IsInstanceValid((Node)ts));
```

Comment:
@Kasterisk you wrote this system, right? So could you comment on how possible it is that there is an *invalid* object in here?
I personally think this should just be a `HasQueuedTransitions and currently running transitions` (however that latter thing could be checked).

Reply:
I've slightly based this upon the `CancelQueuedTransitions` method and tested the functionality with a debugger, although I wasn't able to test all cases.

Reply:
The transitions queue free themselves after finished which is why there's an invalid object in `TransitionSequence`. In a way usage of `IsInstanceValid` works as a check for any active transitions or currently running transitions in a set sequence.

An alternative in this case that I could think of (haven't tested) without any need for IsInstanceValid is turning the local variable at line 160 to a private member variable maybe as currentTransition, and so you can just make the in progress check as `HasQueuedTransitions && currentTransition != null`. This variable can later be cleared in the if block at line 152 when the sequence finishes.

Reply:
Why don't the objects remove themselves from the sequence when they are done, then? That seems much better than keeping disposed objects around.

Reply:
Yes that's also preferable. I just didn't find a good way to do that back when implementing the class.

Reply:
I've used this kind of approach:
```
                    if (runSteps.Peek().RunStep(results))
                        runSteps.Dequeue();
```
to remove an object once it was done. A similar thing could probably be adapted here so that the transitions tell the main code when to remove it.

Reply:
Now that you mention `Dequeue`, I forgot to mention that `TransitionSequence` is a list and just a helper to capture `queuedTransition`'s initial sequence structure. `queuedTransitions` is the actual queue that's used for advancing through the transition sequence which also removes the current transition from itself. So maybe we can use `queuedTransitions` Count for the in progress check?

Reply:
Why there's actually multiple of those lists?
Wouldn't it be simpler if we had a method that requires you to give it a list of actions already, which would immediately go to the running things?
That would solve the problem as well, or?

Reply:
> Why there's actually multiple of those lists?

It's mainly due to CancelQueuedTransitions. I think I remember the problem when implementing it was that queudTransitions doesn't include the currently running transition (because of Dequeue) so what happen was all subsequent transitions got skipped but not the one that's currently running, the solution I came up with was the `TransitionSequence` to hold references to all transition objects from initial queuedTransitions...

You know, I now realize it's a bit silly. In hindsight it probably makes most sense to go with the `currentTransition` variable I mentioned earlier and then just iterate through queuedTransitions to skip all remaining transitions without needing an extra list, so only one queuedTransitions needed, probably will be more robust as well.

> Wouldn't it be simpler if we had a method that requires you to give it a list of actions already, which would immediately go to the running things?

Yes, the current system works a bit like that. We first give TransitionManager "a list" of actions by adding to the transitions queue by calling either `AddScreenFade` or `AddCutscene` and then we call StartTransitions which will immediately advance through the sequence.

Reply:
So, is this line fine? 😅

Reply:
I say this is not because "IsInstanceValid" is a huge design smell to me. This PR needs to be designed some other way but I think we can leave the overall approach changing to @Kasterisk

Reply:
I think for now it's fine to use `HasQueuedTransitions`. It kind of already functions as `HasTransitionInProgress` and is relevant until sequence is finished and the list is cleared. Any further will probably warrant a medium rework as mentioned in earlier discussions.